transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -279,6 +280,11 @@ public int registeredChannels() {
         return channels.size();
     }
 
+    @Override
+    public Iterator<AbstractEpollChannel> registeredChannelsIterator() {
```

Comment:
If channels map is empty we can avoid allocations and return empty iterator instead